src/Parallel/centrality/pagerank.jl

```diff
@@ -8,30 +8,33 @@ function pagerank(
     # indegree(g, v) is estimated run-time to iterate over inneighbors(g, v)
     partitions = LightGraphs.optimal_contiguous_partition(indegree(g), nthreads(), nv(g))
 
-    # collect dangling nodes
-    dangling_nodes = [v for v in vertices(g) if outdegree(g, v) == 0]
+    α_div_outdegree = Vector{Float64}(undef,nv(g))
+    dangling_nodes = Vector{U}()
+    for v in vertices(g)
+        if outdegree(g, v) == 0
+            push!(dangling_nodes, v)
+        end
+        α_div_outdegree[v] = (α/outdegree(g, v))
+    end
+
     N = Int(nv(g))
     # solution vector and temporary vector
-    x = fill(1.0 / N, N)
+    x::Vector{Float64} = fill(1.0 / N, N)
```

Comment:
why do we need a type assertion here?

Reply:
For some reason the compiler infers the return type Any instead of Float64

Reply:
That's the infamous closure performance bug.
Try using a `let` block around the multi-threaded loop.
Here's an [example](https://github.com/JuliaGraphs/LightGraphs.jl/blob/master/src/Parallel/traversals/gdistances.jl).

Reply:
This is interesting,fixing it for now but I'll have to read up on it.

Reply:
I always get bitten by that!